### PR TITLE
Changes for the addition of a new dglc component 

### PR DESCRIPTION
**Pull request by mvertens**
### Description of changes
Updates for adding a cdeps dglc component

### Specific notes

Contributors other than yourself, if any: None

CMEPS Issues Fixed:  None

Are changes expected to change answers? bfb 

Any User Interface Changes: Only if are using the new dglc component, otherwise no.

### Testing performed
The testing was done with the new dglc component on derecho.
The changes should not effect any other existing CESM configuration.
I can run a separate test suite if needed.

### CODE REVIEW

File: mediator/med_phases_profile_mod.F90 [hunk 1]
```diff
@@ -170,7 +170,8 @@ subroutine med_phases_profile(gcomp, rc)
           call ESMF_TimeGet(nexttime, timestring=nexttimestr, rc=rc)
           if (med_utils_ChkErr(rc,__LINE__,u_FILE_u)) return
           ! get current wall clock time
-          call ESMF_TimeSet(wallclockTime, calkindflag=ESMF_CALKIND_GREGORIAN, rc=rc)
+          ! s=0 is to prevent an internal divide by 0 error in esmf
```
Comment (jedwards4b) [inline mediator/med_phases_profile_mod.F90:173]: @mvertens I think that a further optimization is possible here.   Make wallclocktime a saved variable and move lines 174-175 to the first iteration only - line 120.   I'm out today or I would try it myself.

**Comment by mvertens** [comment]
@jedwards4b - thanks for bringing this up. I'm on travel until 4/8. I can look at this on 4/9.

**Comment by mvertens** [comment]
@jedwards4b - can we please merge this PR now and deal with the optimization later. We really need this in order for the dglc capability to work. I'm happy to work on the optimization issue as soon as possible.

**Comment by jedwards4b** [comment]
I actually have that change in another PR - no worries.

**Pull request merged**